### PR TITLE
Implement Forem Organization Unified Embed

### PR DESCRIPTION
**Pull request by msarit**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR implements the [unified embed tag](https://github.com/forem/forem/issues/15099) for Forem Organization embeds.

## Related Tickets & Documents
closes https://github.com/forem/forem/issues/15567

## QA Instructions, Screenshots, Recordings
- As a user, create an organization if there isn't one. Next, create a Forem Organization liquid tag using this format: `{% embed <forem-org-url> %}`. NOTE: this is the organization's Forem-specific url (`forem-domain/org-slug`), NOT the organization's actual website URL. The Liquid Tag should render properly.
- Also create a Forem Org liquid tag using the old method: `{% organization <forem-org-slug> %}` or `{% org <forem-org-slug> %}`. This should also work, as I am leaving the old implementation in place for now.

### UI accessibility concerns?
None

## Added/updated tests?
- [X] Yes

## [Forem core team only] How will this change be communicated?
- [X] I will share this change internally with the appropriate teams
Considering the project as a whole, I shall update the Liquid Tag documentation at the end. Since I am not changing the current implementation, the present documentation is still valid. Plus, I will avoid updating the docs piecemeal, and risk confusing users.

## [optional] Are there any post deployment tasks we need to perform?
None

### CODE REVIEW

**Comment by djuber** [comment]
Test cases

```markdown
## Embed style

{% embed http://localhost:3000/org8606 %}

## classic style

{% organization org938 %}

{% org org7767 %}
```

Result
Screenshot from 2022-01-14 11-37-13

File: app/liquid_tags/organization_tag.rb [hunk 1]
```diff
@@ -22,7 +23,15 @@ def render(_context)
   end
 
   def parse_slug_to_organization(organization)
-    organization = Organization.find_by(slug: organization)
+    if organization.include?(URL.url.to_s)
```
Comment (djuber) [inline app/liquid_tags/organization_tag.rb:26]: clarity: this check says "did this link point to the site?" - is there a better way to do this than calling URL.url? it took a second to determine what this said, I'm just looking for something that says to the reader "this is the site's url" in an unambiguous way.

**Review by djuber — APPROVED** [review]

File: app/liquid_tags/organization_tag.rb (hunk 1 above)
Comment (djuber) [reply app/liquid_tags/organization_tag.rb:26]: nitpick: https://github.com/forem/forem/blob/main/app/lib/url.rb#L19 we already cast this to string before returning (URL.url returns the string representation of the site) - the `to_s` call is unneeded.

Comment (djuber) [reply app/liquid_tags/organization_tag.rb]: should this say `organization.starts_with?(URL.url)`

File: app/liquid_tags/organization_tag.rb [hunk 2]
```diff
@@ -23,7 +23,7 @@ def render(_context)
   end
 
   def parse_slug_to_organization(organization)
-    forem_domain = URL.url.to_s
+    forem_domain = URL.url
```
Comment (djuber) [inline app/liquid_tags/organization_tag.rb:26]: thanks!

File: app/liquid_tags/organization_tag.rb (hunk 1 above)
Comment (msarit) [reply app/liquid_tags/organization_tag.rb]: I guess `starts_with` is more specific 🤷🏾‍♀️  Ok I will change

File: app/liquid_tags/organization_tag.rb [hunk 3]
```diff
@@ -22,7 +23,16 @@ def render(_context)
   end
 
   def parse_slug_to_organization(organization)
-    organization = Organization.find_by(slug: organization)
+    forem_domain = URL.url
+    if organization.starts_with?(forem_domain)
```
Comment (djuber) [inline app/liquid_tags/organization_tag.rb:27]: I always get this method name wrong! 

https://ruby-doc.org/core-3.1.0/String.html#method-i-start_with-3F

Sorry for the misleading suggestion

**Review by jeremyf — APPROVED** [review]

**Pull request merged**